fpsim/arrays.py

```diff
@@ -0,0 +1,151 @@
+            # People/Sim are initialized
+            self.len_used = 0
+            self.len_tot = 0
+            self.initialized = skip_init
+            self.raw = np.empty((0,self.ncols), dtype=dtype)
+        else:
+            # This Arr is a temporary object used for intermediate calculations when we want to index an array
+            # by UID (e.g., inside an update() method). We allow this state to reference an existing, initialized
+            # People object, but do not register it for dynamic growth
+            self.len_used = self.people.uid.len_used
+            self.len_tot = self.people.uid.len_tot
+            self.initialized = True
+            self.raw = np.full(shape=(self.len_tot, self.ncols), dtype=self.dtype, fill_value=self.nan)
+
+        return
+
+    def _convert_key(self, key):
+        """
+        Used for getitem and setitem to determine whether the key is indexing
+        the raw array (``raw``) or the active agents (``values``), and to convert
+        the key to array indices if needed.
+        """
+        if isinstance(key, (ss.uids, int, ss.dtypes.int)):
+            return key
+        elif isinstance(key, (ss.BoolArr, ss.IndexArr)):
+            return key.uids
+        elif isinstance(key, slice):
+            return self.auids[key]
+        elif not np.isscalar(key) and len(key) == 0: # Handle [], np.array([]), etc.
+            return ss.uids()
+        elif isinstance(key, np.ndarray) and ss.options.reticulate:
+            return key.astype(int)
+        else:
+            errormsg = f'Indexing an Arr ({self.name}) by ({key}) is ambiguous or not supported. Use ss.uids() instead, or index Arr.raw or Arr.values.'
+            raise Exception(errormsg)
+
+    def __getitem__(self, key):
+        if isinstance(key, (slice, tuple)):
+            index = self._convert_key(key[0])
+            return self.raw[index, key[1]]
+        else:
+            index = self._convert_key(key)
+            return self.raw[index, :]
+
+    def __setitem__(self, key, value):
+        if isinstance(key, (slice, tuple)):
+            index = self._convert_key(key[0])
+            self.raw[index, key[1]] = value
+        return
+
+    def __gt__(self, other): raise NotImplementedError("Comparison not supported for TwoDimensionalArray")
+    def __lt__(self, other): raise NotImplementedError("Comparison not supported for TwoDimensionalArray")
+    def __ge__(self, other): raise NotImplementedError("Comparison not supported for TwoDimensionalArray")
+    def __le__(self, other): raise NotImplementedError("Comparison not supported for TwoDimensionalArray")
+    def __eq__(self, other): raise NotImplementedError("Comparison not supported for TwoDimensionalArray")
+    def __ne__(self, other): raise NotImplementedError("Comparison not supported for TwoDimensionalArray")
+
+    def count(self):
+        raise NotImplementedError("count() is not implemented for TwoDimensionalArray.")
```

Comment:
Can they be sliced? Would it make sense to allow something like `count(self, axis)`?